src/Reflection/ParameterReflection.php

```diff
@@ -39,15 +43,13 @@ public function getDeclaringClass()
     #[ReturnTypeWillChange]
     public function getClass()
     {
-        $phpReflectionType = parent::getType();
-        if ($phpReflectionType === null) {
+        $type = parent::getType();
+
+        if (! ($type instanceof ReflectionNamedType && ! $type->isBuiltin())) {
```

Comment:
Had to double check this. Maybe the following would be quicker to review at a glance?

```suggestion
        if (! $type instanceof ReflectionNamedType || $type->isBuiltin())) {
```

Reply:
You applied De Morgan's law on this, basically :D 

Works for me, will apply after testing locally

Reply:
66064ca32631c3888466fe8f555a71793b0cac52